dcicutils/s3_utils.py

```diff
@@ -394,6 +354,125 @@ def unzip_s3_to_s3(self, zipped_s3key, dest_dir, acl=None, store_results=True):
+            raise ValueError(f"Missing {self.LEGACY_ES_URL_KEY!r} or {self.ES_URL_KEY!r}"
+                             f" key in global_env {env_description}.")
+
+        self._env_name = (env_description.get(self.LEGACY_ENV_NAME_KEY) or
+                          env_description.get(self.ENV_NAME_KEY))
+        if not self._env_name:
+            raise ValueError(f"Missing {self.LEGACY_ENV_NAME_KEY!r} or {self.ENV_NAME_KEY!r}"
+                             f" key in global_env {env_description}.")
+
+    @property
+    def portal_url(self):
+        return self._portal_url
+
+    @property
+    def es_url(self):
+        return self._es_url
+
+    @property
+    def env_name(self):
+        return self._env_name
+
+    @staticmethod
+    def verify_and_get_env_config(s3_client, global_bucket: str, env):
+        """ Verifies the S3 environment from which the env config is coming from, and returns the S3-based env config
+            Throws exceptions if the S3 bucket is unreachable, or an env based on the name of the global S3 bucket
+            is not present.
+        """
+        logger.warning('Fetching bucket data via global env bucket: {}'.format(global_bucket))
+        head_response = s3_client.head_bucket(Bucket=global_bucket)
+        status = head_response['ResponseMetadata']['HTTPStatusCode']  # should be 200; raise error for 404 or 403
+        if status != 200:
+            raise GlobalBucketAccessError(global_bucket=global_bucket, status=status)
+        # list contents of global env bucket, look for a match with the global env bucket name
+        list_response = s3_client.list_objects_v2(Bucket=global_bucket)
+        # no match, raise exception
+        if list_response['KeyCount'] < 1:
+            raise CannotInferEnvFromNoGlobalEnvs(global_bucket=global_bucket)
+        keys = [content['Key'] for content in list_response['Contents']]
+        if env is None:
+            if len(keys) == 1:
+                # If there is only one env, which is the likely case, let's infer that this is the one we want.
+                env = keys[0]
+                logger.warning(f"No env was specified, but {env} is the only one available, so using that.")
+            else:
+                raise CannotInferEnvFromManyGlobalEnvs(global_bucket=global_bucket, keys=keys)
+        if env not in keys:
+            raise MissingGlobalEnv(global_bucket=global_bucket, keys=keys, env=env)
+        # we found a match, so fetch that file as config
+        get_response = s3_client.get_object(Bucket=global_bucket, Key=env)
+        env_config = json.loads(get_response['Body'].read())
+        return env_config
+
+    @staticmethod
+    def fetch_health_page_json(url, use_urllib):
+        # TODO: Does anyhone know what problem this use_urllib thing was solving?  Was it solving some bug
+        #       in Python 2 that no longer matters? These two solutions should be equivalent. Is there a way
+        #       to simplify this? -kmp 22-Aug-2021
```

Comment:
`requests` does not work (give it a try). No idea why.

Reply:
I don't think it's that requests doesn't work - just that there is no json representation returned for the health page (same sort of thing for indexing_status and perhaps some other endpoints?

And in that case maybe the whole requests clause should just be removed and always use urllib?

Reply:
Well, there _is_ a JSON representation returned when `?format=json` is given, or there's supposed to be. But I guess you're right if it fails to do that, that could be the problem. It's possible there is someplace in which tool is defaulting or failing to default a `Content-Type` that is confusing things. Anyway, I removed the version that does `requests.get` and left the one that does `urllib.request.urlopen`.